File: app/src/main/java/ua/in/quireg/anothermovieapp/managers/MovieTrailersProvider.java
package ua.in.quireg.anothermovieapp.managers;

import android.content.Context;
import android.net.Uri;
import android.util.Log;

import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import ua.in.quireg.anothermovieapp.common.MovieItem;
import ua.in.quireg.anothermovieapp.common.MovieTrailer;
import ua.in.quireg.anothermovieapp.common.UriHelper;
import ua.in.quireg.anothermovieapp.movieDetails.FetchTrailersCallback;

public final class MovieTrailersProvider {

    private static final String LOG_TAG = MovieTrailersProvider.class.getSimpleName();

    public static void fetchTrailersList(MovieItem movie, Context c,
                                         final FetchTrailersCallback callback) {

        final Uri uri = UriHelper.getMovieTrailerUriById(String.valueOf(movie.getId()));

        JsonObjectRequest movieTrailersRequest = new JsonObjectRequest(
                Request.Method.GET, uri.toString(), null, new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response) {

                try {
                    JSONArray arr = response.getJSONArray("results");
                    List<MovieTrailer> trailers = new ArrayList<>();
                    for (int i = 0; i < arr.length(); i++) {
                        JSONObject obj = arr.getJSONObject(i);

                        MovieTrailer movieTrailer = new MovieTrailer(
                                obj.getString("id"),
                                obj.getString("iso_639_1"),
                                obj.getString("iso_3166_1"),
                                obj.getString("key"),
                                obj.getString("name"),
                                obj.getString("site"),
                                obj.getString("size"),
                                obj.getString("type")
                        );
                        trailers.add(movieTrailer);
                    }
                    callback.onTrailersFetchCompleted(trailers);
                } catch (JSONException e) {
                    Log.w(LOG_TAG, e.getMessage());
                }

            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Log.d(LOG_TAG, "Failed: " + uri.toString());
            }
        });
        Log.d(LOG_TAG, "Fetching: " + uri.toString());
        Volley.newRequestQueue(c).add(movieTrailersRequest);
    }
}

File: app/src/main/java/com/anothermovieapp/managers/MovieTrailersProvider.kt
package com.anothermovieapp.managers

import android.content.Context
import android.util.Log
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import com.anothermovieapp.repository.Movie
import com.anothermovieapp.repository.MovieTrailer
import com.anothermovieapp.common.UriHelper
import com.anothermovieapp.movieDetails.FetchTrailersCallback
import org.json.JSONException
import java.util.*

object MovieTrailersProvider {
    private val LOG_TAG = MovieTrailersProvider::class.java.simpleName
    fun fetchTrailersList(movie: Movie?, c: Context?, callback: FetchTrailersCallback) {
        val uri = UriHelper.getMovieTrailerUriById(movie?.id.toString())
        val movieTrailersRequest = JsonObjectRequest(
                Request.Method.GET, uri.toString(), null, Response.Listener { response ->
            try {
                val arr = response.getJSONArray("results")
                val trailers: MutableList<MovieTrailer> = ArrayList()
                for (i in 0 until arr.length()) {
                    val obj = arr.getJSONObject(i)
                    val movieTrailer = MovieTrailer(
                            obj.getString("id"),
                            obj.getString("iso_639_1"),
                            obj.getString("iso_3166_1"),
                            obj.getString("key"),
                            obj.getString("name"),
                            obj.getString("site"),
                            obj.getString("size"),
                            obj.getString("type")
                    )
                    trailers.add(movieTrailer)
                }
                callback.onTrailersFetchCompleted(trailers)
            } catch (e: JSONException) {
                Log.w(LOG_TAG, e.message)
            }
        }, Response.ErrorListener { Log.d(LOG_TAG, "Failed: $uri") })
        Log.d(LOG_TAG, "Fetching: $uri")
        Volley.newRequestQueue(c).add(movieTrailersRequest)
    }
}